### DISCUSSION TITLE
Chain Selector in the "Connect your wallet" modal

### DISCUSSION THREAD

**Discussion by claude-eric-steiner** [WalletConnect/walletconnect-monorepo#3117] [Ideas]
When freshly connecting to a wallet, the WalletConnect modal "Connect your wallet" pops up.

Even when setting up WalletConnect to allow to connect to multiple chains, 
there is no chain selector in that modal. Only after connecting, there is the possibility to change the chain in the "Select network" modal.
(cool feature by the way!)

This missing chain selector on the "Connect your wallet" modal results often (though not always) in the inability to connect with the QR code to a specific wallet, and also directly to a mobile wallet app when a wrong default chain is set, and the wallet is not accepting this chain.

Therefore it would make a lot of sense to have the chain selector (same enumeration view like in the "Select network" modal) already available in the "Connect your wallet" modal. This way the default chain would be set according to user input, and the chances to correctly connect to a wallet app (with QR code or directly to an installed app) would be greatly increased.

Many dapps go multi chain nowadays, and a feature for the user to select the chain of desire **before** he connects is really missing.

Any thoughts?

Cheers Stoney

**Comment by glitch-txs** [comment]
Thanks Claude, Web3Modal has already this feature where user can select a required chain for the proposal and the rest of the chains will be set to optional, you can get this by calling the open function with the 'selectNetwork' param instead of ConnectWallet, this would require you to make a custom button though.
[docs on this topic](https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/hooks#reference)

**Comment by claude-eric-steiner** [comment]
Thank you Glitch, 
this sounds interesting, need to play with it to better understand. 
As I imagine, it is still a separate modal handling and needs the creation of a custom button (like you say), 
and I still think, to have the chain selector (optionally) included in the connect modal would be a better user experience.
Something like this:
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/17033493/9aff6afd-fc6e-45d4-b52d-5e04a165d4fa)

Just my 5 cents...
